packages/auth-construct/src/lambda/reference_auth_initializer.test.ts

```diff
@@ -0,0 +1,517 @@
+            throw awsSDKErrorMessageMock;
+          }
+          return listIdentityProvidersResponse;
+        }
+        if (request instanceof DescribeUserPoolClientCommand) {
+          if (describeUserPoolClientResponse.$metadata.httpStatusCode !== 200) {
+            throw awsSDKErrorMessageMock;
+          }
+          return describeUserPoolClientResponse;
+        }
+        if (request instanceof ListGroupsCommand) {
+          if (listGroupsResponse.$metadata.httpStatusCode !== 200) {
+            throw awsSDKErrorMessageMock;
+          }
+          return listGroupsResponse;
+        }
+        return undefined;
+      }
+    );
+    mock.method(
+      identityClient,
+      'send',
+      async (
+        request: DescribeIdentityPoolCommand | GetIdentityPoolRolesCommand
+      ) => {
+        if (request instanceof DescribeIdentityPoolCommand) {
+          if (describeIdentityPoolResponse.$metadata.httpStatusCode !== 200) {
+            throw awsSDKErrorMessageMock;
+          }
+          return describeIdentityPoolResponse;
+        }
+        if (request instanceof GetIdentityPoolRolesCommand) {
+          if (getIdentityPoolRolesResponse.$metadata.httpStatusCode !== 200) {
+            throw awsSDKErrorMessageMock;
+          }
+          return getIdentityPoolRolesResponse;
+        }
+        return undefined;
+      }
+    );
+  });
+  void it('handles create events', async () => {
+    const result = await handler.handleEvent(createCfnEvent);
+    assert.deepEqual(result.Status, 'SUCCESS');
+    assert.equal(
+      result.PhysicalResourceId,
+      '00000000-0000-0000-0000-000000000000'
+    );
+    assert.deepEqual(result.Data, expectedData);
+  });
+
+  void it('handles update events', async () => {
+    const result = await handler.handleEvent(updateCfnEvent);
+    assert.deepEqual(result.Status, 'SUCCESS');
+    assert.deepEqual(result.Data, expectedData);
+  });
+
+  void it('handles delete events', async () => {
+    const result = await handler.handleEvent(deleteCfnEvent);
+    assert.deepEqual(result.Status, 'SUCCESS');
+  });
+
+  void it('throws if fetching user pool fails', async () => {
+    describeUserPoolResponse = httpError;
+    await rejectsAndMatchError(
+      handler.handleEvent(createCfnEvent),
+      awsSDKErrorMessageMock.message
+    );
+  });
+
+  void it('throws if fetching user pool fails', async () => {
+    describeUserPoolResponse = {
+      ...httpSuccess,
+      UserPool: undefined,
+    };
+    await rejectsAndMatchError(
+      handler.handleEvent(createCfnEvent),
+      'Failed to retrieve the specified UserPool.'
+    );
+  });
+
+  void it('throws if user pool has no password policy', async () => {
+    describeUserPoolResponse = {
+      ...httpSuccess,
+      UserPool: {
+        ...UserPool,
+        Policies: undefined,
+      },
+    };
+    await rejectsAndMatchError(
+      handler.handleEvent(createCfnEvent),
+      'Failed to retrieve password policy.'
+    );
+  });
```

Comment:
One idea to consider in this file is to have parametrized test. The look very similar and seem to differ by input and output. (not a blocker but something worth trying).

Example: https://github.com/aws-amplify/amplify-backend/blob/b6761b02b0640d2cce9c465b2d824c60f08ddebd/scripts/components/api-changes-validator/api_usage_generator.test.ts#L356-L357